### ISSUE TITLE
Workers should not take jobs until after the `OnApplicationBootstrap` event

### ISSUE THREAD

**Issue by epechuzal** [nestjs/bull#1664]
### Is there an existing issue that is already proposing this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe it

I notice that workers on the `@nestjs/bullmq` module immediately start work after their `OnModuleInit` call. This means that they can start work before all other necessary modules are initialized.

### Describe the solution you'd like

When using NestJS as a server the app will not start listening for API calls until after the `OnApplicationBootstrap` event.

`BullModule` should behave in a similar way: workers should not start picking jobs from queues until after this event.

### What is the motivation / use case for changing the behavior?

This change would make NestJS bull behavior more consistent with other parts of the NestJS ecosystem.

**Comment by kamilmysliwiec** [comment]
If there's a self-containing module that provides workers, wouldn't you expect these workers to be ready & start processing jobs upon the `onModuleInit` hook? What I'm trying to illustrate here is that it really comes down to how you interpret `onModuleInit` and `onApplicationBootstrap` hooks - and this may differ depending on several factors

**Comment by epechuzal** [comment]
@kamilmysliwiec That same logic would also apply to API controllers but we still provide a `onApplicationBootstrap` hook before they accept connections. 

If the problem is interpretation then I would like to request a flag we can provide to the module not to start taking jobs until the bootstrap event (e.g. add a `waitUntilBootstrap: boolean` flag to the `BullModuleOptions` interface?).

**Comment by kamilmysliwiec** [comment]
> If the problem is interpretation then I would like to request a flag we can provide to the module not to start taking jobs until the bootstrap event (e.g. add a waitUntilBootstrap: boolean flag to the BullModuleOptions interface?).

I'd be more than happy to approve this PR! The steps to implement this should be pretty straightforward if you're interested in contributing to this one!:

- update interface
- navigate to `bull.explorer` and add the `onApplicationBootstrap` hook below `onModuleInit`
- depending on the flag run the corresponding lifecycle hook

**Comment by epechuzal** [comment]
@kamilmysliwiec So I looked at the code and I couldn't find a "nice" way to accomplish what I am trying to do here, then I stumbled on `WorkerOpts` which has `autorun: boolean`, and figured that the functionality I am looking for is already covered, maybe?

```ts
@Processor('my-queue', { autorun: false })
export class MyWorker extends WorkerHost implements OnApplicationBootstrap {
  async onApplicationBootstrap(): Promise<void> {
    if (!this.worker.isRunning()) {
      this.worker.run();
    }
  }
}
```

This code runs on my project and doesn't appear to cause issues.  I will close this issue as my personal problem is solved 

Normally I'd say this is worth documenting but maybe my use-case is too niche for the official doc, I will leave that up to you :)

**Comment by Dracks** [comment]
hi @epechuzal,

I don't think your case is so niche case. You cannot have bullmq with workers runing with the cqrs module. Since nestjs/cqrs module registers the handlers onApplicationBootstrap. 

Tbh: I will be happy to make a P/R with the changes proposed.
